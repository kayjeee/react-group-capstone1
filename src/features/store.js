import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketSlice';
import missionReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
  },
});

export default store;
