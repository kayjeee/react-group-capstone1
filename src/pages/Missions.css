/* Colors */
:root {
  --border-color: #ccc;
  --error-color: red;
  --primary-color: #999;
  --secondary-color: lightseagreen;
  --success-color: green;
}

/* Table Styles */
.missionTable {
  position: relative;
  border-collapse: collapse;
}

.missionTable::before,
.missionTable::after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border: 1px solid var(--border-color);
  pointer-events: none;
}

.missionTable::before {
  z-index: 1;
}

.missionTable thead {
  background-color: #ddd;
}

.missionTable tr:nth-child(even) {
  background-color: #eee;
}

.missionTable th,
.missionTable td {
  border: none;
  padding: 0.5rem;
}

/* Badge Styles */
.status-badge {
  background: var(--primary-color);
  border-radius: 0.25rem;
  color: white;
  display: block;
  font-size: 0.65rem;
  font-weight: bold;
  min-width: 8rem;
  padding: 0.15rem;
  text-align: center;
  text-transform: uppercase;
}

.status-badge.joined {
  background-color: var(--secondary-color);
}

/* Action Button Styles */
.action-button {
  border: 1px solid #777;
  border-radius: 0.5rem;
  color: var(--success-color);
  cursor: pointer;
  display: block;
  font-size: 0.85rem;
  margin-top: 1rem;
  min-width: 8rem;
  padding: 0.25rem;
  text-align: center;
}

.action-button.leave-button {
  border-color: var(--error-color);
  color: var(--error-color);
}

.action-button.join-button {
  border-color: var(--success-color);
  color: var(--success-color);
}
