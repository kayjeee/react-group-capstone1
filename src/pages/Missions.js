import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions, joinMission, leaveMission } from '../redux/features/missions/missions';
import './Missions.css';

export default function Missions() {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!missions.length) dispatch(fetchMissions());
  }, []);

  return (
    <section className="container">
      <h2>Missions</h2>
      <table className="missionTable">
        <thead>
          <tr>
            <th>Name</th>
            <th>Description</th>
            <th>Status</th>
            <th>Action</th> {/* New column for active member and leave mission buttons */}
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.id}>
              <td>{mission.name}</td>
              <td className="font-sm">{mission.description}</td>
              <td className="mission-status-cell">
                {!mission.joined && (
                  <>
                    <span className="status-badge">Not a member</span>
                  </>
                )}
                {mission.joined && (
                  <>
                    <span className="status-badge joined">Active Member</span>
                  </>
                )}
              </td>
              <td className="mission-action-cell"> {/* New column for active member and leave mission buttons */}
                {!mission.joined && (
                  <button
                    type="button"
                    className="action-button join-button"
                    onClick={() => dispatch(joinMission(mission.id))}
                  >
                    Join Mission
                  </button>
                )}
                {mission.joined && (
                  <button
                    type="button"
                    className="action-button leave-button"
                    onClick={() => dispatch(leaveMission(mission.id))}
                  >
                    Leave Mission
                  </button>
                )}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </section>
  );
}
