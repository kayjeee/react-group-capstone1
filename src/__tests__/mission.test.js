import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../features/store';
import Missions from '../components/missions';

describe('User Interactions test', () => {
  it('will test user interaction', async () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
