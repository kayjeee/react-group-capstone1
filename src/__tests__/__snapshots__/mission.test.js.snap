// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Interactions test will test user interaction 1`] = `
<section
  className="container"
>
  <table
    className="missionTable"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</section>
`;
